File: code/lambda.java
public class SomeJavaClass {

   interface Addable {
       int add(int a, int b);
   }

   static int add2(int a, int b, Addable addable) {
       if (a > b) {
           return addable.add(a, b);
       } else {
           return a - b;
       }
   }

   void test() {
       System.out.println(add2(2, 1, new Addable() {
           @Override
           public int add(int a, int b) {
               return a + b;
           }
       }));
   }
}

File: code/lambda.kt
fun f3(a: Int, b: Int, fn: (Int, Int) -> Int): Int =
       if (a > b) fn(a, b) else a - b

fun test() {
   f3(2, 1) { a1, a2 ->
       a1 + a2
   }                               // => 3       
   f3(2, 1, { a1, a2 -> a1 + a2})  // => 3
}